aiven/datasource_service_component.go

```diff
@@ -115,11 +115,11 @@ func datasourceServiceComponentRead(d *schema.ResourceData, m interface{}) error
 			// check optional ssl search criteria, if not set by a user match entries
 			// without ssl or ssl=true
 			if ssl, ok := d.GetOk("ssl"); ok {
-				if c.Ssl != strconv.FormatBool(ssl.(bool)) {
+				if *c.Ssl != ssl.(bool) {
```

Comment:
`c.Ssl` might be `nil` if someone specifies ssl attribute for service component that doesn't have it on server side. Skipping over would be better than nil pointer dereference.